packages/hardhat-core/src/internal/hardhat-network/provider/miner/edr.ts

```diff
@@ -77,13 +81,15 @@ export class EdrMiner implements BlockMinerAdapter {
         if ("pc" in traceItem) {
           await vmTracer.addStep(traceItem);
 
-          // For solidity-coverage compatibility
-          this._vmStub.evm.events.emit("step", {
-            pc: Number(traceItem.pc),
-            depth: traceItem.depth,
-            opcode: { name: traceItem.opcode },
-            stackTop: traceItem.stackTop,
-          });
+          for (const listener of this._stepListeners) {
+            await listener({
```

Comment:
Can the listeners throw? I'm also wondering if we should add some warning for slow listeners or alternatively call them outside of the call stack with `setTimeout(listener, 0)` to avoid blocking here.

Reply:
Hmm, yes, in theory they can throw. But I think we can ignore that for now; I don't think the behavior is fundamentally different to what was happening before (and this should be temporary anyway).